python-avd/pyavd/_eos_cli_config_gen/schema/__init__.py

```diff
@@ -127,15 +127,13 @@ class Methods(AvdList[MethodsItem]):
                 Methods._item_type = MethodsItem
 
                 _fields: ClassVar[dict] = {"type": {"type": str}, "methods": {"type": Methods}}
-                type: Literal["none", "start-stop", "stop-only"]
+                type: Type
```

Comment:
we may one day have class with something else maybe.
We could try prefixing / suffixing with `T_` / `_T` to be safe (and not go mad :) )

Reply:
what if this shadows `Literal` ? or some other class - changing the meaning inside the class

Reply:
for instance
```
+  literal:
+    type: str
+    valid_values: ["break-stuff"]

```

generates

```
     WanRouteServers._item_type = WanRouteServersItem

+    Literal: TypeAlias = Literal["break-stuff"]
+

```

Reply:
Discussed on dev call with @ClausHolbechArista - we acknowledge this is a corner case that could occur but for now we shall live with it - it has never bitten us for the nested class names so far.

#YOLOPS